apollo-rxsupport/src/main/java/com.apollographql.android.rx/RxApollo.java

```diff
@@ -26,65 +25,47 @@ private RxApollo() {
   }
 
   @Nonnull
-  public static <T> Observable<T> from(@Nonnull
-  final ApolloWatcher<T> watcher) {
+  public static <T> Observable<T> from(@Nonnull final ApolloWatcher<T> watcher) {
     checkNotNull(watcher);
-    return Observable.create(new Observable.OnSubscribe<T>() {
-      @Override public void call(final Subscriber<? super T> subscriber) {
-        cancelOnUnsubscribe(subscriber, watcher);
+    return Observable.fromEmitter(new Action1<Emitter<T>>() {
+      @Override public void call(final Emitter<T> emitter) {
+        emitter.setCancellation(new Cancellable() {
+          @Override public void cancel() throws Exception {
+            watcher.cancel();
+          }
+        });
         watcher.enqueueAndWatch(new ApolloCall.Callback<T>() {
           @Override public void onResponse(@Nonnull Response<T> response) {
-            if (!subscriber.isUnsubscribed()) {
-              subscriber.onNext(response.data());
-            }
+            emitter.onNext(response.data());
           }
 
           @Override public void onFailure(@Nonnull Throwable e) {
             Exceptions.throwIfFatal(e);
-            if (!subscriber.isUnsubscribed()) {
-              subscriber.onError(e);
-            }
+            emitter.onError(e);
           }
         });
-
       }
-    });
+    }, Emitter.BackpressureMode.LATEST);
```

Comment:
Should this be taken as a parameter? 

Would there be cases where the user would want to buffer unconsumed `onNext` calls instead?